api/python/provisioner/commands/check.py

```diff
@@ -609,7 +677,255 @@ def _cluster_status(*, args: str) -> CheckEntry:
+            return check_entry
+
+        res: List[CheckEntry] = list()
+        _IP_DEV_IFACE_TMPL = "ip addr show dev {ifc} | grep 'inet' | grep '{ifc}'"
+
+        for node in nodes:
+            check_ret: CheckEntry = CheckEntry(cfg.Checks.PUB_DATA_IP.value)
+
+            try:
+                # Get list data network interfaces
+                iface = Check._get_pillar_data(
+                    f"cluster/{node}/network/data_nw/iface")
+
+                ifc = [ifc for ifc in iface if 's0f0' in ifc]
+
+                if ifc:
+                    # Get IP address assigned to interface
+                    cmd = _IP_DEV_IFACE_TMPL.format(ifc=ifc[0])
+                    _ip = cmd_run(cmd, targets=node,
+                                  fun_kwargs=dict(python_shell=True))
+
+                    # Extract IP address
+                    ip = _ip[node].split()[1].split('/')[0]
+
+                    # Check for connectivity of received IP
+                    NetworkV().validate('connectivity', [ip])
+                    check_ret.set_passed(checked_target=node)
+                else:
+                    check_ret.set_fail(checked_target=node,
+                                        comment="Public data interface " \
+                                                "'s0f0' not found. iface "\
+                                                f"available: {iface}")
+            except Exception as exc:
+                check_ret.set_fail(checked_target=node, comment=str(exc))
+
+            res.append(check_ret)
+
+        return res
+
+    @staticmethod
+    def _controller_ip(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
+        """
+        Check controller ips
+
+        :param args: controller_ip check specific parameters and arguments
+        :return:
+        """
+        check_entry: CheckEntry = CheckEntry(cfg.Checks.CONTROLLER_IP.value)
+
+        if cortx_py_utils_import_error:
+            check_entry.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment="Package cortx-py-utils not installed")
+            return check_entry
+
+        try:
+            # Get node list
+            nodes = Check._get_pillar_data("cluster/node_list")
+        except Exception as exc:
+            check_entry.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            return check_entry
+
+        res: List[CheckEntry] = list()
+
+        for node in nodes:
+            check_ret: CheckEntry = CheckEntry(cfg.Checks.CONTROLLER_IP.value)
+
+            try:
+                primary_mc_ip = Check._get_pillar_data(
+                                "storage_enclosure/controller/primary_mc/ip")
+                secondary_mc_ip = Check._get_pillar_data(
+                                "storage_enclosure/controller/secondary_mc/ip")
+
+                # Check for connectivity of received IPs
+                NetworkV().validate('connectivity',
```

Comment:
I see that different parameters are used for validators: 'connectivity', 'luns', 'lvms'. Maybe it is better to create enum/constants for that. And I think it is better to create them in validation framework and import then.

Reply:
Sure. I will create separate PR to address this for all checks.

Reply:
Ok thank you. Please, create also jira ticket with improvement tag for that.